### PR TITLE
Make ManagementHandler more easily extensible 

### PR DESCRIPTION
**Pull request by ajnyga**
https://github.com/pkp/pkp-lib/issues/4819

### CODE REVIEW

**Comment by ajnyga** [comment]
Can not see where the conflict is? @asmecher @NateWr

**Comment by NateWr** [comment]
This can happen when the commit hashes are not the same, even if there is not a "real" conflict on pkp-lib. It's one of the annoying things about submodules. :/

The fix is to rebase pkp-lib, roll back the submodule commit in ojs, rebase ojs, redo the submodule commit.

**Comment by ajnyga** [comment]
do you `git rebase master` in pkp-lib?

**Comment by NateWr** [comment]
```
cd <ojs-root>
cd lib/pkp
git checkout master
git pull upstream master
git checkout <your-branch>
git rebase -i upstream/master
git push --force

cd ../..
git checkout master
git pull upstream master
git checkout <your-branch>
git reset HEAD~1
git rebase -i upstream/master
git add lib/pkp
git commit -m "...your-submodule-commit-message-..."
git push --force
```

**Comment by ajnyga** [comment]
ok thanks, I have a feeling that in the past doing checkout master in pkp-lib and then doing pull request lead to a situation where my own pr included commits from other users. So I've used to doing checkout -b something right after doing the submodule update in OJS so that I use the version of pkp-lib that OJS fetches for the submodule update.

**Comment by ajnyga** [comment]
It probably actually worked! There is some hope with git afterall!

File: pages/management/SettingsHandler.inc.php [hunk 1]
```diff
@@ -37,19 +37,26 @@ function __construct() {
...
 
 		$templateMgr = TemplateManager::getManager($request);
 		$context = $request->getContext();
 		$router = $request->getRouter();
 		$dispatcher = $request->getDispatcher();
 
+		// Hook into the settings templates to add the appropriate tabs
+		HookRegistry::register('Template::Settings::distribution', function($hookName, $args) {
+			$templateMgr = $args[1];
+			$output = &$args[2];
```
Comment (asmecher) [inline pages/management/SettingsHandler.inc.php]: Preferred syntax is `$a =& $b` rather than `$a = &$b`

Comment (ajnyga) [reply pages/management/SettingsHandler.inc.php]: check

**Comment by ajnyga** [comment]
hmm after fixing the cosmetic intend problems @asmecher notice the tests now fail with dependency related stuff

**Comment by asmecher** [comment]
Sorry for dropping the ball on this one, @ajnyga. The commits list still shows two submodule updates for lib/pkp -- There Can Be Only One, and it needs to be the last on the list.

**Comment by ajnyga** [comment]
@asmecher see https://github.com/pkp/ojs/pull/2427

**Pull request closed without merging**